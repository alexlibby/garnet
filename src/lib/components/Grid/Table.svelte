<svelte:options customElement="garnet-table" />

<script>
  import Grid from "./Grid.svelte";
  import Cell from "./Cell.svelte";

  export let columnCount = 4;
  export let rowCount = 4;
  export let itemCount = 0;
  export let border = "";
  export let displayImages = false;
  export let label = "Test Label";
  export let imageLabel = "untitled";
  export let displayImageLabels = false;
</script>   

<h1>{label}</h1>
<Grid columns={columnCount} rows={rowCount} {border}>
  {#each { length: itemCount } as _, i}
    <Cell 
      {displayImages} 
      imageLabel={imageLabel} 
      displayImageLabels={displayImageLabels}
    />
  {/each}
</Grid> 

<style>
  h1 {
    font-family: Arial, Helvetica, sans-serif;
    border-bottom: 0.8px solid rgba(38, 85, 115, 0.15);
  }
</style>