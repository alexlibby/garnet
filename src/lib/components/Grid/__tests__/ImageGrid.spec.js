import { describe, test, expect } from "vitest";
import { axe } from "vitest-axe";
import * as AxeMatchers from "vitest-axe/matchers";
import { render, screen, fireEvent } from "@testing-library/svelte";
import Table from "../../Grid/Table.svelte";

const GridHTML = `<div class="garnet-grid s--9EA4qNh9tr7"><div class="s--9EA4qNh9tr7" style="grid-template-rows: repeat(4, 1fr); grid-template-columns: repeat(1, auto); border: none;"><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--><div class="cell s-wR0iEIfZdiFJ"></div><!--<Cell>--></div></div>`;
expect.extend(AxeMatchers);

describe("Tests for ImageGrid", () => {
  const props = {
    columnCount: "1",
    rowCount: "4",
    border: "none",
    displayImages: false,
    itemCount: "12",
    displayImageLabels: false,
  };

  test("should render properly without images", async () => {
    const result = render(Table, { props });
    expect(() => result).not.toThrow();
  });

  test("should render properly if images added", async () => {
    const result = render(Table, { ...props, displayImages: true });
    expect(() => result).not.toThrow();
  });

  test("should render properly if images added with text labels", async () => {
    const result = render(Table, {
      ...props,
      displayImages: true,
      displayImageLabels: true,
    });

    const imageCount = result.getAllByTestId("placeholderImage");
    const labelCount = result.getAllByTestId("placeholderLabel");

    expect(imageCount.length).toBe(12);
    expect(labelCount.length).toBe(12);
  });

  test("get a snapshot of component", () => {
    // Create a new container for the test and mount component
    const host = document.createElement("div");
    document.body.appendChild(host);
    const instance = new Table({ target: host, props: props });

    // check that all rendered OK
    expect(instance).toBeTruthy();

    // Take snapshot
    expect(host.innerHTML).toMatchSnapshot();
  });

  test("should demonstrate no issues with accessibility", async () => {
    const render = () => GridHTML;
    const result = render();

    expect(await axe(result)).toHaveNoViolations();
  });
});
