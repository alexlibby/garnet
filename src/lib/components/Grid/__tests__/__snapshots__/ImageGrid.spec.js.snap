// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for ImageGrid > get a snapshot of component 1`] = `"<h1 class="svelte-1nqlrak">Test Label</h1> <div class="garnet-grid svelte-5nvth"><div style="grid-template-rows: repeat(4, 1fr); grid-template-columns: repeat(1, auto);" class="svelte-5nvth"><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div><div class="cell svelte-z6p2xb"></div></div></div>"`;
